### DISCUSSION TITLE
How to change UI state, onChange of <Select> value?

### DISCUSSION THREAD

**Discussion by spidertwin2** [react-hook-form/react-hook-form#4963] [Q&A] [answered]
Hello,

On change of `<Select>` value, I want to change value of `<InputLeftAddon>`.  If I select Fiji, it should show +679 instead of +123.

[Link](https://codesandbox.io/s/hungry-feistel-hgkz4?file=/components/PhoneForm.tsx) to sandbox.

Thanks  
:turtle:

**Comment by bluebill1049** [comment, marked as answer]
You will have to move `register` before render

```ts
const { onChange, ...rest } = register('test')

<input onChange={(e) => {
  onChange(e.target.value);
  
  // setValue('otherField', 'whatevervalue');
}} />

<input {...register('otherField')} />
```

**Reply by spidertwin2** [reply]
Thanks @bluebill1049 
:smile:

**Reply by bluebill1049** [reply]
Welcome 🤗

**Comment by spidertwin2** [comment]
Because I am not changing the value of a registered input field, I added useState to change the value of InputLeftAddon.  

But `onChange(e.target.value)` is giving me the error - 
Cannot read property 'value' of undefined.

Please see sandbox for updated code.

**Reply by bluebill1049** [reply]
`onChange(e)` try that

**Reply by spidertwin2** [reply]
Works! :palm_tree:
